### PR TITLE
[WIP] 6413/integrate endpoints for post scheduling

### PR DESCRIPTION
**Pull request by kirrg001**
#6413

Integrate endpoints for post scheduling and handle permissions for clients
- added endpoints for publish scheduled post and get all scheduled posts
- from/to option for get scheduled posts
- frontend can fetch scheduled posts in admin (tested)
- remove client type check in auth middleware, because it brings no value!
### TODO
- [x] wait for subscriber branch (both subscriber branch and i integrated a handling for handling client context ----> we use external)

### CODE REVIEW

File: core/server/models/plugins/filter.js [hunk 1]
```diff
@@ -162,7 +164,38 @@ filter = function filter(Bookshelf) {
...
+            return this;
+        },
+
+        /**
+         * Apply custom filters only
+         *
+         * @param {object} options
+         * @returns {Model}
+         */
+        applyCustomFilters: function applyCustomFilters(options) {
```
Comment (kirrg001) [inline core/server/models/plugins/filter.js]: TODO: write a unit test

File: core/server/permissions/effective.js [hunk 2]
```diff
@@ -7,6 +8,10 @@ effective = {
     user: function (id) {
         return Models.User.findOne({id: id, status: 'all'}, {include: ['permissions', 'roles', 'roles.permissions']})
             .then(function (foundUser) {
+                if (!foundUser) {
+                    return Promise.reject(new Error('incorrect usage, no user found'));
```
Comment (kirrg001) [inline core/server/permissions/effective.js]: use errors....

File: core/server/models/base/index.js [hunk 3]
```diff
@@ -133,12 +133,12 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
 
     // Get the user from the options object
     contextUser: function contextUser(options) {
-        // Default to context user
         if (options.context && options.context.user) {
             return options.context.user;
-        // Other wise use the internal override
         } else if (options.context && options.context.internal) {
             return 1;
+        } else if (options.context && options.context.client) {
```
Comment (kirrg001) [inline core/server/models/base/index.js]: external return 0

File: package.json [hunk 4]
```diff
@@ -73,6 +73,7 @@
     "pg": "4.1.1"
   },
   "devDependencies": {
+    "async": "2.0.0-rc.3",
```
Comment (ErisDS) [inline package.json]: Is it possible to drop this? I think the [sequence util](https://github.com/TryGhost/Ghost/blob/master/core/server/utils/sequence.js) does the same as  `async.waterfall` 😁

Comment (kirrg001) [inline package.json]: thanks for hint. the difference between ```async.waterfall``` and sequence is that your sequence util helper expects that the functions in the array return a promise. setup and teardown are right now callback designed, but i changed it to support promise and callback usage. with this change i am able to use sequence.

File: core/server/models/base/index.js (hunk 3 above)
Comment (kirrg001) [inline core/server/models/base/index.js]: will wait that subscribe branch is merged into master, then i will simply delete this condition.

**Comment by kirrg001** [comment]
closed because of #6861

**Pull request closed without merging**